Java/core/src/main/java/com/amazon/randomcutforest/AbstractForestUpdateExecutor.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.randomcutforest;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public abstract class AbstractForestUpdateExecutor<P> {
+
+    protected final IUpdateCoordinator<P> updateCoordinator;
+    protected final ArrayList<IUpdatable<P>> models;
+
+    protected AbstractForestUpdateExecutor(IUpdateCoordinator<P> updateCoordinator, ArrayList<IUpdatable<P>> models) {
+        this.updateCoordinator = updateCoordinator;
+        this.models = models;
+    }
+
+    public long getTotalUpdates() {
+        return updateCoordinator.getTotalUpdates();
+    }
+
+    /**
+     * Update the forest with the given point. The point is submitted to each
+     * sampler in the forest. If the sampler accepts the point, the point is
+     * submitted to the update method in the corresponding Random Cut Tree.
+     *
+     * @param point The point used to update the forest.
+     */
+    public void update(double[] point) {
```

Comment:
question. if a client is using float[] for compact rcf, should it also take float[] input?

Reply:
The repository is still in an intermediate state, with some classes using double[] and others (mostly the new ones) using float[]. Now that we've agreed to release this branch as a 2.0 version, let's address this inconsistency in a separate, targeted PR.